### PR TITLE
Enable Python 3.5

### LINKED ISSUE #1616: Python 3.5 does not work with privacyIDEA
Currently privacyIDEA does not work with Python 3.5, which is the default python 3 version in Ubuntu 16.04.
The `json.loads()` function expects a unicode string in python 3.5. In later versions, unicode and byte strings are supported.
The Flask-Response `data`-attribute, which we use to feed `json.loads()`, is unfortunately a byte string.
We could work around this problem by
1. converting all data passed to `json.loads()` to unicode first (which is done in most of the tests) or
2. use the Flask-Response `get_json()` function.

### PR DESCRIPTION
**Pull request by plettich**
privacyIDEA did not work correctly with python 3.5 due to some json
encoding issues.
Since we are using Flask 1.0, the Response class provides a json property
which we can use to read the corresponding data.
To avoid caching of the json content, a new Response-class disables the caching.

Fixes #1616

### CODE REVIEW

**Comment by plettich** [comment]
Remarks from #1762 and #1706 are integrated.

File: privacyidea/lib/eventhandler/base.py [hunk 1]
```diff
@@ -299,7 +297,7 @@ def _get_tokenowner(request):
     @staticmethod
     def _get_response_content(response):
         if response:
-            content = json.loads(response.data)
+            content = response.get_json(cache=False, force=True)
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/lib/eventhandler/base.py]: fixed in 67700324

File: privacyidea/app.py [hunk 2]
```diff
@@ -91,6 +92,16 @@
 }
 
 
+class PiResponseClass(Response):
+    @property
+    def json(self):
+        """This will contain the parsed JSON data if the mimetype indicates
+        JSON (:mimetype:`application/json`, see :meth:`is_json`), otherwise it
+        will be ``None``.
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/app.py]: fixed in 67700324

[the remark this answers is not in the document]
Comment (fredreichbier) [reply privacyidea/app.py]: thanks!

**Pull request merged**